File: app/src/main/java/com/mackenzie/downhub/ui/component/binding/BottomNavigationViewBinding.java
package com.mackenzie.downhub.ui.component.binding;

import androidx.databinding.BindingAdapter;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.mackenzie.downhub.R;

public class BottomNavigationViewBinding {

    @BindingAdapter("app:selectedItemId")
    public static void setSelectedItemId(BottomNavigationView view, int position) {
        int id;
        switch (position) {
            case 0: id = R.id.tab_home; break;
            case 1: id = R.id.tab_browser; break;
            case 2: id = R.id.tab_progress; break;
            case 3: id = R.id.tab_video; break;
            default: id = R.id.tab_settings; break;
        }
        view.setSelectedItemId(id);
    }
}

File: app/src/main/java/com/mackenzie/ui/component/binding/BottomNavigationViewBinding.kt
package com.mackenzie.ui.component.binding

import androidx.databinding.BindingAdapter
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.mackenzie.R

object BottomNavigationViewBinding {

    @BindingAdapter("app:selectedItemId")
    @JvmStatic
    fun BottomNavigationView.setSelectedItemId(position: Int) {
        selectedItemId = when (position) {
            0 -> R.id.tab_browser
            1 -> R.id.tab_progress
            2 -> R.id.tab_video
            else -> R.id.tab_settings
        }
    }
}